### PR TITLE
Rebuild new notice and choice 23 version

### PR DESCRIPTION
**Pull request by deathcave**
### Related Ticket(s)

[Noitce & Choice w3 Publisher](https://w3.ibm.com/w3publisher/urx/notice-and-choice-v23)

### Description

Notice & Choice is a legally required component to create a form where IBM collects its customer information.
This section shows the specific product's legal links, terms, and conditions. Along with it, it collects users' consent regarding communication preferences.

### Changelog

**New**

-  A new react component to create notice & choice user interface.

**Changed**

-  Notice and Choice content loaded by languages.

**Removed**

- Postal option removed from new notice and choice version

### CODE REVIEW

**Comment by deathcave** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by deathcave** [comment]
recheck

**Comment by deathcave** [comment]
recheck

**Comment by deathcave** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by deathcave** [comment]
recheck

File: packages/web-components/src/components/notice-choice/notice-choice.ts [hunk 1]
```diff
@@ -159,14 +151,11 @@ class NoticeChoice extends StableSelectorMixin(LitElement) {
     }
   }
   countryChangeAction() {
-    const splitValue = this.locale.split('-', 2);
+    const splitValue = this.locale;
     const ncData = getNcContentFromWindow();
```
Comment (sanjitbauli) [inline packages/web-components/src/components/notice-choice/notice-choice.ts:155]: unused variable ncData

File: packages/web-components/src/components/notice-choice/notice-choice.ts [hunk 2]
```diff
@@ -159,14 +151,11 @@ class NoticeChoice extends StableSelectorMixin(LitElement) {
...
-    if (splitValue[1] === 'en') {
+    if (splitValue == 'en') {
       let preText = this.defaultPreText;
```
Comment (sanjitbauli) [inline packages/web-components/src/components/notice-choice/notice-choice.ts:157]: unused variable preText

File: packages/web-components/src/components/notice-choice/notice-choice.ts [hunk 3]
```diff
@@ -184,24 +173,7 @@ class NoticeChoice extends StableSelectorMixin(LitElement) {
     const cc = newVal.toLocaleLowerCase();
```
Comment (sanjitbauli) [inline packages/web-components/src/components/notice-choice/notice-choice.ts:173]: unused variable cc

File: packages/web-components/src/components/notice-choice/notice-choice.ts [hunk 4]
```diff
@@ -184,24 +173,7 @@ class NoticeChoice extends StableSelectorMixin(LitElement) {
...
     const lc = countrySettings[newVal.toLocaleLowerCase()].lang;
```
Comment (sanjitbauli) [inline packages/web-components/src/components/notice-choice/notice-choice.ts:174]: unused variable lc

File: packages/web-components/src/components/notice-choice/notice-choice.ts [hunk 5]
```diff
@@ -308,86 +278,96 @@ class NoticeChoice extends StableSelectorMixin(LitElement) {
...
+  }
   postTextTemplate() {
     if (this.ncData) {
-      const OptInContent = this.ncData.OptInContent;
-      const OtherPreferences = this.ncData.OtherPreferences;
-      let postText = OptInContent.postText;
+      const OtherPreferences = this.ncData.trialPrivacyText;
       let tcHtml = '';
+      let postText = this.ncData.postText;
+
```
Comment (sanjitbauli) [inline packages/web-components/src/components/notice-choice/notice-choice.ts:353]: unused variable tcHtml

**Review by sanjitbauli — CHANGES_REQUESTED** [review]

**Comment by deathcave** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by deathcave** [comment]
recheck

**Comment by deathcave** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by deathcave** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by deathcave** [comment]
recheck

**Comment by deathcave** [comment]
I have read the DCO document and I hereby sign the DCO.

**Comment by deathcave** [comment]
recheck

**Comment by sanjitbauli** [comment]
@kennylam - following up with the blocker around DCO Assistant and other checks. thanks

**Comment by kennylam** [comment]
> **DCO Assistant Lite bot:** Thanks for your submission! We ask that you all sign our [Developer Certificate of Origin](https://github.com/carbon-design-system/carbon-dco/blob/main/dco.md) before we can accept your contribution. You can sign the DCO by adding a comment below using this text:
> 
> _**I have read the DCO document and I hereby sign the DCO.**_
> 
> **1** out of **2** committers have signed the DCO.✅
> @kennylam❌ @rajaneesh22
> You can retrigger this bot by commenting **recheck** in this Pull Request

@deathcave Did you change your user name since creating this PR? The DCO assistant is needs a signature from `@rajaneesh22`.

**Comment by deathcave** [comment]
recheck

**Comment by kennylam** [comment]
@deathcave Trying to sign with your current user name will not work. It's looking for `@rajaneesh22`. I suggest you close this PR and open a new one.

**Pull request closed without merging**